addons/website_blog/controllers/main.py

```diff
@@ -287,10 +286,11 @@ def blog_post_copy(self, blog_post_id, **post):
         return werkzeug.utils.redirect("/blog/%s/post/%s?enable_editor=1" % (slug(new_blog_post.blog_id), slug(new_blog_post)))
 
     @http.route('/blog/post_change_background', type='json', auth="public", website=True)
-    def change_bg(self, post_id=0, cover_properties={}, **post):
-        if not post_id:
-            return False
-        return request.env['blog.post'].browse(int(post_id)).write({'cover_properties': json.dumps(cover_properties)})
+    def change_bg(self, blog_id, post_id=0, cover_properties={}, **post):
+        if blog_id:
```

Comment:
hmmm... not tested but at first view, condition is probably in wrong order...

if you have a post_id you write on it, else your write on blog

And like first comment, 

```python
if x:
    model, id = 'post', int(post_id)
else:
    model, id = 'blog', int(blog_id)
```

eturn request.env[model].browse(id).write({'cover_properties': json.dumps(cover_properties)})